src/wasm/wasm-binary.cpp

```diff
@@ -295,7 +295,8 @@ void WasmBinaryWriter::writeFunctionSignatures() {
 }
 
 void WasmBinaryWriter::writeExpression(Expression* curr) {
-  ExpressionStackWriter<WasmBinaryWriter>(curr, *this, o, debug);
+  BinaryenIRToBinaryWriter writer(*this, o);
+  writer.visit(curr);
```

Comment:
this and similar things can be `BinaryenIRToBinaryWriter(*this, o).visit(curr);` I think, on one line?

Reply:
Done. Also fixed a few more similar places.